icgcget/cli.py

```diff
@@ -71,6 +55,38 @@ def logger_setup(logfile):
     return logger
 
 
+def check_code(client, code):
+    if code != 0:
+        logger.error("{} client exited with a nonzero error code {}.".format(client, code))
+        raise click.ClickException("Please check client output for error messages")
```

Comment:
You may want to create your own exception objects / classes that represent the various failure modes of the lower levels (clients)